moto/rds/models.py

```diff
@@ -845,8 +862,8 @@ def vpc_security_group_ids(
         self._vpc_security_group_ids = vpc_security_group_ids
 
     @property
-    def domain_memberships(self) -> List[str]:
-        return []
```

Comment:
This was originally just a dummy method, so it got the type hint wrong.  No need to keep `list[str]` around as an option.  Also, don't return `None`.  AWS always returns an empty list if there are no memberships.